torch/nn/parallel/distributed.py

```diff
@@ -319,6 +319,15 @@ class DistributedDataParallel(Module, Joinable):
         :class:`torch.distributed.optim.DistributedOptimizer` for optimizing
         parameters.
 
+    .. note::
+        DistributedDataParallel currently offers limited support for gradient
+        checkpointing with :meth:`torch.utils.checkpoint`. DDP will work as
+        expected when there are no unused parameters in the model and each layer
+        is checkpointed at most once (make sure you are not passing
+        `find_unused_parameters=True` to DDP). We currently do not support the
+        case where a layer is checkpointed multiple times, or when there unused
+        parameters in the checkpointed model.
```

Comment:
This is supported with `static_graph` though right? Maybe we should have an additional comment that if your graph is static this can still be supported?

Reply:
It is supported with static graph but it is currently a private API which we haven't documented. My thinking is that in general we shouldn't recommend the use of private APIs in docs and shouldn't say the feature is supported unless we have a public OSS API for it that is in beta state, what do you think?

Reply:
Ah I see, I didn't realize static graph is still prototype.